x/concentrated-liquidity/position_test.go

```diff
@@ -937,6 +937,208 @@ func (s *KeeperTestSuite) TestHasAnyPosition() {
 	}
 }
 
+func (s *KeeperTestSuite) TestCreateFullRangePosition() {
+	s.Setup()
+	defaultAddress := s.TestAccs[0]
+	DefaultJoinTime := s.Ctx.BlockTime()
+	defaultPositionCoins := sdk.NewCoins(DefaultCoin0, DefaultCoin1)
+
+	tests := []struct {
+		name                  string
+		remainingLockDuration time.Duration
+		isLocked              bool
+		isUnlocking           bool
+	}{
+		{
+			name: "full range position",
+		},
+		{
+			name:                  "full range position: locked",
+			remainingLockDuration: 24 * time.Hour * 14,
+			isLocked:              true,
+		},
+		{
+			name:                  "full range position: unlocking",
+			remainingLockDuration: 24 * time.Hour,
+			isUnlocking:           true,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		s.Run(test.name, func() {
+			// Init suite for each test.
+			s.Setup()
+			s.Ctx = s.Ctx.WithBlockTime(DefaultJoinTime)
+
+			// Create a default CL pools
+			clPool := s.PrepareConcentratedPool()
+
+			var (
+				positionId         uint64
+				liquidity          sdk.Dec
+				concentratedLockId uint64
+				err                error
+			)
+
+			// Fund the owner account
+			s.FundAcc(defaultAddress, defaultPositionCoins)
+
+			// System under test
+			if test.isLocked {
+				positionId, _, _, liquidity, _, concentratedLockId, err = s.App.ConcentratedLiquidityKeeper.CreateFullRangePositionLocked(s.Ctx, clPool, defaultAddress, defaultPositionCoins, test.remainingLockDuration)
+			} else if test.isUnlocking {
+				positionId, _, _, liquidity, _, concentratedLockId, err = s.App.ConcentratedLiquidityKeeper.CreateFullRangePositionUnlocking(s.Ctx, clPool, defaultAddress, defaultPositionCoins, test.remainingLockDuration)
+			} else {
+				positionId, _, _, liquidity, _, err = s.App.ConcentratedLiquidityKeeper.CreateFullRangePosition(s.Ctx, clPool, defaultAddress, defaultPositionCoins)
+			}
+
+			s.Require().NoError(err)
+
+			// Check position
+			_, err = s.App.ConcentratedLiquidityKeeper.GetPosition(s.Ctx, positionId)
+			s.Require().NoError(err)
+
+			// Check lock
+			if test.isLocked || test.isUnlocking {
+				concentratedLock, err := s.App.LockupKeeper.GetLockByID(s.Ctx, concentratedLockId)
+				s.Require().NoError(err)
+				s.Require().Equal(liquidity.TruncateInt().String(), concentratedLock.Coins[0].Amount.String())
```

Comment:
Can we also check whether it's locked or unlocking?

Reply:
added https://github.com/osmosis-labs/osmosis/pull/4883/commits/6030a18469e5de91dac4fce3faa4e5dc6c17425a